examples/gno.land/r/demo/foo20/foo20.gno

```diff
@@ -1,131 +1,99 @@
-	if err != nil {
-		panic(err)
-	}
-	return allowance
+	ownerAddr := users.Resolve(owner)
+	spenderAddr := users.Resolve(spender)
+	return token.Allowance(ownerAddr, spenderAddr)
 }
 
-// setters.
-
 func Transfer(to pusers.AddressOrName, amount uint64) {
-	caller := std.PrevRealm().Addr()
-	err := foo.Transfer(caller, users.Resolve(to), amount)
-	if err != nil {
-		panic(err)
-	}
+	toAddr := users.Resolve(to)
+	checkErr(token.Transfer(toAddr, amount))
 }
 
 func Approve(spender pusers.AddressOrName, amount uint64) {
-	caller := std.PrevRealm().Addr()
-	err := foo.Approve(caller, users.Resolve(spender), amount)
-	if err != nil {
-		panic(err)
-	}
+	spenderAddr := users.Resolve(spender)
+	checkErr(token.Approve(spenderAddr, amount))
 }
 
 func TransferFrom(from, to pusers.AddressOrName, amount uint64) {
-	caller := std.PrevRealm().Addr()
-	err := foo.TransferFrom(caller, users.Resolve(from), users.Resolve(to), amount)
-	if err != nil {
-		panic(err)
-	}
+	fromAddr := users.Resolve(from)
+	toAddr := users.Resolve(to)
+	checkErr(token.TransferFrom(fromAddr, toAddr, amount))
 }
 
-// faucet.
-
+// Faucet is distributing foo20 tokens without restriction (unsafe).
+// For a real token faucet, you should take care of setting limits are asking payment.
 func Faucet() {
-	// FIXME: add limits?
-	// FIXME: add payment in gnot?
 	caller := std.PrevRealm().Addr()
-	err := foo.Mint(caller, 1000*10000) // 1k
-	if err != nil {
-		panic(err)
-	}
+	amount := uint64(1_000 * 10_000) // 1k
+	checkErr(banker.Mint(caller, amount))
 }
 
-// administration.
-
-func Mint(address pusers.AddressOrName, amount uint64) {
-	caller := std.PrevRealm().Addr()
-	assertIsAdmin(caller)
-	err := foo.Mint(users.Resolve(address), amount)
-	if err != nil {
-		panic(err)
-	}
+func Mint(to pusers.AddressOrName, amount uint64) {
+	admin.AssertCallerIsOwner()
+	toAddr := users.Resolve(to)
+	checkErr(banker.Mint(toAddr, amount))
 }
 
-func Burn(address pusers.AddressOrName, amount uint64) {
-	caller := std.PrevRealm().Addr()
-	assertIsAdmin(caller)
-	err := foo.Burn(users.Resolve(address), amount)
-	if err != nil {
-		panic(err)
-	}
+func Burn(from pusers.AddressOrName, amount uint64) {
+	admin.AssertCallerIsOwner()
+	fromAddr := users.Resolve(from)
+	checkErr(banker.Burn(fromAddr, amount))
 }
 
-// render.
-//
-
 func Render(path string) string {
 	parts := strings.Split(path, "/")
 	c := len(parts)
 
 	switch {
 	case path == "":
-		return foo.RenderHome()
+		return banker.RenderHome()
 	case c == 2 && parts[0] == "balance":
 		owner := pusers.AddressOrName(parts[1])
-		balance, _ := foo.BalanceOf(users.Resolve(owner))
+		ownerAddr := users.Resolve(owner)
+		balance := banker.BalanceOf(ownerAddr)
 		return ufmt.Sprintf("%d\n", balance)
 	default:
 		return "404\n"
 	}
 }
 
-func assertIsAdmin(address std.Address) {
-	if address != admin {
-		panic("restricted access")
+func checkErr(err error) {
+	if err != nil {
+		panic(err)
```

Comment:
shouldn't it be `panic(err.Error())` ??
previous code was panicking with `string` value.

Reply:
Both works, I prefer panic-inducing error types when possible.